src/applications/auth/containers/AuthApp.jsx

```diff
@@ -130,11 +133,21 @@ export class AuthApp extends React.Component {
     const authMetrics = new AuthMetrics(type, payload);
     authMetrics.run();
     setupProfileSession(authMetrics.userProfile);
-    this.redirect();
+    this.redirect(authMetrics.userProfile);
   };
 
-  redirect = () => {
+  redirect = (userProfile = {}) => {
```

Comment:
@hinzed1127 I might be overthinking this, but is it at all possible to move this logic back to the standalone sign in page?  For example could we make the `returnUrl` either a string or a callback, and if the later execute the function to determine the logic?

Reply:
I don't think a callback would work, since `sessionSession` stores everything as strings, so parsing that out would be painful. 

The main rub is preserving the `application` and `to` parameters. Since we lose those as soon as we redirect away, we need to store somewhere. Right now that's in `verify`, because I didn't want to muddy up the logic in `standaloneRedirect` when we should only have to worry about this scenario for calls to `verify`.

However, when someone is redirected back to `auth/login/callback` after being logged in, the check in this code is just `if there's a returnUrl defined in sessionStorage then redirect to that`. This situation introduced a case where we don't want to get rid of that returnUrl, and instead force them onto the `/sign-in/verify` page. What it seemed to come down to was the need to add an exception to the redirect logic _somewhere_, and this seemed to be the be the least intrusive to the callback pattern.

Reply:
@hinzed1127 and I tried refactoring the code and putting all the logic in the VerifyPage component, it worked, but it made the experience slower for the user because it had to render the AuthApp page, then the VerifyPage before redirecting the user.  I don't think consolidating the logic is worth the slower experience for the user.  @hinzed1127 thanks again for walking me through the experiment.